tests/tracer/test_processors.py

```diff
@@ -216,3 +217,76 @@ def test_aggregator_partial_flush_2_spans():
     assert writer.pop() == [child1, child2]
     parent.finish()
     assert writer.pop() == [parent]
+
+
+def test_trace_top_level_span_processor_same_service_name():
+    """Parent span and child span have the same service name"""
+    trace_processors = TraceTopLevelSpanProcessor()
+    parent = Span(None, "parent", service="service")
+    parent._local_root = parent
+
+    child1 = Span(None, "child1", service="service")
+    child1.parent_id = parent.span_id
+    child1._parent = parent
+    trace_processors.process_trace([parent, child1])
+    # parent span should be set to top_level
+    # child span should not contain top_level metric
+    assert parent.get_metric("_dd.top_level") == 1
+    assert "_dd.top_level" not in child1.metrics
+
+
+def test_trace_top_level_span_processor_different_service_name():
+    """Parent span and child span have the different service names"""
+    trace_processors = TraceTopLevelSpanProcessor()
+    parent = Span(None, "parent", service="service")
+    parent._local_root = parent
+
+    child2 = Span(None, "child2", service="new_service_name")
+    child2.parent_id = parent.span_id
+    child2._parent = parent
+    trace_processors.process_trace([parent, child2])
+    # child span should be set to top_level
+    assert parent.get_metric("_dd.top_level") == 1
+    assert child2.get_metric("_dd.top_level") == 1
+
+
+def test_trace_top_level_span_processor_orphan_span():
+    """Trace chuck does not contain parent span"""
+    trace_processors = TraceTopLevelSpanProcessor()
+    parent = Span(None, "span_not_in_trace", service="service")
+    parent._local_root = parent
+
+    orphan_span = Span(None, "orphan_span", service="service")
+    orphan_span._parent = parent
+    orphan_span.parent_id = orphan_span._parent.span_id
+    trace_processors.process_trace([Span(None, "span1"), orphan_span])
```

Comment:
is `Span(None, "span1")` needed here?

Reply:
Nope I just added it here to ensure that the trace processor does not process just the first span in a trace but iterates to the end. I could remove it if it makes this test confusing

Reply:
that should be a separate test case.

for example, if someone were to remove this later because it doesn't seem related to testing of the orphan span, then you would lose your validation that process_trace iterates through all of the spans.

we can try to keep each unit test case to fully test one specific behavior. if we have 10 scenarios we want to test, we should have 10 tests, try not to merge them together.